arb_os/gasAccounting.mini

```diff
@@ -1251,21 +1174,21 @@ public func deferredPayments_new() -> DeferredPayments {
     return stack_new();
 }
 
-public func deferredPayments_add(dp: DeferredPayments, recipient: address, amount: uint) -> DeferredPayments {
-    return stack_push(dp, (recipient, amount));
+public func deferredPayments_add(dp: DeferredPayments, escrowKey: uint, recipient: address, amount: uint) -> DeferredPayments {
+    return stack_push(dp, (escrowKey, recipient, amount));
 }
 
 public func deferredPayments_apply(acctStore: AccountStore, dp: DeferredPayments) -> AccountStore {
     return xloop {
         if let Some(res) = stack_pop(dp) {
             let (udp, item) = res;
             dp = udp;
-            let (recipient, amount) = unsafecast<(address, uint)>(item);
-            acctStore = accountStore_set(
-                acctStore,
-                recipient,
-                account_addToEthBalance(accountStore_get(acctStore, recipient), amount),
-            );
+            let (escrowKey, recipient, amount) = unsafecast<(uint, address, uint)>(item);
+            if let Some(uAcctStore) = accountStore_payFromEscrow(acctStore, escrowKey, recipient, amount) {
+                acctStore = uAcctStore;
+            } else {
+                asm(11000,) { debugprint };
```

Comment:
Is it possible this will hit and if so would it make sense to revert?

Reply:
Changed this to have the function return an option, then dealt with the failure case appropriately at each callsite.  (For all of the current callsites, ignoring the failure is the best approach.)